src/service/chat.js

```diff
@@ -206,6 +206,58 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
+};
+
+const initChatFromPostAnon = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await anonymousApi.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `initChatFromPostAnon` function is almost identical to the `initChatFromPost` function, except that it uses `anonymousApi` instead of `api`. This repetition could be avoided by creating a more generic function that accepts the API instance as a parameter. Here's how you could refactor these two functions:

```diff
- const initChatFromPost = async ({source, id}) => {
+ const initChatFromPost = async ({source, id}, apiInstance) => {
    let payload = {source};
    switch (source) {
      case 'post':
        payload = {...payload, postId: id};
        break;
      case 'comment':
        payload = {...payload, commentId: id};
        break;
      default:
        break;
    }

    try {
      const response = await apiInstance.post('/chat/init-chat-from-post', payload);
      if (response?.status === 200) {
        return Promise.resolve(response.data);
      }

      return Promise.reject(response.data);
    } catch (e) {
      if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
      return Promise.reject(e);
    }
  };

- const initChatFromPostAnon = async ({source, id}) => {
-   let payload = {source};
-   switch (source) {
-     case 'post':
-       payload = {...payload, postId: id};
-       break;
-     case 'comment':
-       payload = {...payload, commentId: id};
-       break;
-     default:
-       break;
-   }
-
-   try {
-     const response = await anonymousApi.post('/chat/init-chat-from-post', payload);
-     if (response?.status === 200) {
-       return Promise.resolve(response.data);
-     }
-
-     return Promise.reject(response.data);
-   } catch (e) {
-     if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
-     return Promise.reject(e);
-   }
- };
```

Then you can call this function with the appropriate API instance:

```javascript
initChatFromPost({source: 'post', id: 123}, api);
initChatFromPost({source: 'post', id: 123}, anonymousApi);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->